Lean4Axiomatic/Natural/Addition.lean

```diff
@@ -292,4 +293,34 @@ theorem zero_sum_split {n m : ℕ} : n + m ≃ 0 → n ≃ 0 ∧ m ≃ 0 := by
       Rel.trans ‹step (n + m) ≃ step n + m› ‹step n + m ≃ 0›
     exact absurd ‹step (n + m) ≃ 0› step_neqv_zero
 
+
+/-
+Example showing that naturals numbers with addition form a Monoid and use
+that fact to prove something.
+-/
+
+local instance add_monoid_ops :  CA.Monoid.Ops ℕ := {
+  binop := (· + ·)
+  ident := 0
+}
+
+def subst_addR {n₁ n₂ m : ℕ} : n₁ ≃ n₂ → m + n₁ ≃ m + n₂ := AA.substR
+
+def add_monoid_props : CA.Monoid.Props ℕ := {
+  substL   := subst_add
+  substR   := subst_addR
```

Comment:
Since `Substitutive₂` is already defined for naturals, can you just use `AA.substL` and `AA.substR` here? It might require you to first change your group and monoid instances of `Substitutive₂` into _local_ instances.

```suggestion
def subst_addR {n₁ n₂ m : ℕ} : n₁ ≃ n₂ → m + n₁ ≃ m + n₂ := AA.substR

def add_monoid_props : CA.Monoid.Props ℕ := {
  substL   := AA.substL
  substR   := AA.substR
```